pallets/core/src/modules/trust_registry/mod.rs

```diff
@@ -0,0 +1,360 @@
+            + Into<<Self as frame_system::Config>::Event>;
+    }
+
+    /// Stores `TrustRegistry`s information: `Convener`, name, etc.
+    #[pallet::storage]
+    #[pallet::getter(fn trust_registry_info)]
+    pub type TrustRegistriesInfo<T: Config> =
+        StorageMap<_, Blake2_128Concat, TrustRegistryId, TrustRegistryInfo<T>>;
+
+    /// Schema metadata stored in all trust registries. Mapping of the form (schema_id, registry_id) -> schema_metadata
+    #[pallet::storage]
+    #[pallet::getter(fn schema_metadata)]
+    pub type TrustRegistrySchemasMetadata<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        TrustRegistrySchemaId,
+        Blake2_128Concat,
+        TrustRegistryId,
+        TrustRegistrySchemaMetadata<T>,
+    >;
+
+    /// Schema ids corresponding to trust registries. Mapping of registry_id -> schema_id
+    #[pallet::storage]
+    #[pallet::getter(fn registry_schema)]
+    pub type TrustRegistryStoredSchemas<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        TrustRegistryId,
+        Blake2_128Concat,
+        TrustRegistrySchemaId,
+        (),
+    >;
+
+    /// Stores `TrustRegistry`s owned by conveners as a mapping of the form convener_id -> Set<registry_id>
+    #[pallet::storage]
+    #[pallet::getter(fn convener_trust_registries)]
+    pub type ConvenerTrustRegistries<T> =
+        StorageMap<_, Blake2_128Concat, Convener, TrustRegistryIdSet<T>, ValueQuery>;
+
+    /// Stores `Trust Registry`'s `Verifier`s schemas.
+    #[pallet::storage]
+    #[pallet::getter(fn trust_registry_verifier_schemas)]
+    pub type TrustRegistryVerifierSchemas<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        TrustRegistryId,
+        Blake2_128Concat,
+        Verifier,
+        VerifierSchemas<T>,
+        ValueQuery,
+    >;
+
+    /// Stores `Trust Registry`'s `Issuer`s schemas.
+    #[pallet::storage]
+    #[pallet::getter(fn trust_registry_issuer_schemas)]
+    pub type TrustRegistryIssuerSchemas<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        TrustRegistryId,
+        Blake2_128Concat,
+        Issuer,
+        IssuerSchemas<T>,
+        ValueQuery,
+    >;
+
+    /// Stores `Trust Registry`'s `Issuer`s configurations.
+    #[pallet::storage]
+    #[pallet::getter(fn trust_registry_issuer_config)]
+    pub type TrustRegistryIssuerConfigurations<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        TrustRegistryId,
+        Blake2_128Concat,
+        Issuer,
+        TrustRegistryIssuerConfig<T>,
+        ValueQuery,
+    >;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Creates a new `Trust Registry` with the provided identifier.
+        /// The DID signature signer will be set as a `Trust Registry` owner.
+        #[pallet::weight(SubstrateWeight::<T>::init_trust_registry(init_trust_registry, sig))]
+        pub fn init_trust_registry(
```

Comment:
The name `init_trust_registry` seems like it will only create the registry but it can also update.